nucypher/cli/commands/alice.py

```diff
@@ -447,13 +461,57 @@ def grant(general_config,
     # Input validation
     if ALICE.federated_only:
         if any((value, rate)):
-            raise click.BadOptionUsage(option_name="--value, --rate",
-                                       message="Can't use --value or --rate with a federated Alice.")
+            message = "Can't use --value or --rate with a federated Alice."
+            raise click.BadOptionUsage(option_name="--value, --rate", message=message)
     elif bool(value) and bool(rate):
         raise click.BadOptionUsage(option_name="--rate", message="Can't use --value if using --rate")
-    elif not (bool(value) or bool(rate)):
+
+    # Interactive collection follows:
+    # - Label
+    # - Expiration Date & Time
+    # - M of N
+    # - Policy Value (ETH)
+
+    # Label
+    if not label:
+        label = click.prompt(f'Enter label to grant Bob {bob_verifying_key[:8]}', type=click.STRING)
+
+    # Policy Expiration
+    # TODO: Remove this line when the time is right.
+    paint_probationary_period_disclaimer(emitter)
+
+    if not force and not expiration:
+        if ALICE.duration_periods:
+            # TODO: use a default in days or periods?
+            expiration = maya.now() + timedelta(days=ALICE.duration_periods)  # default
+            if not click.confirm(f'Use default policy duration (expires {expiration})?'):
+                expiration = click.prompt('Enter policy expiration datetime', type=click.DateTime())
+        else:
+            # No policy duration default default available; Go interactive
+            expiration = click.prompt('Enter policy expiration datetime', type=click.DateTime())
+
+    # TODO: Remove this line when the time is right.
+    enforce_probationary_period(emitter=emitter, expiration=expiration)
+
+    # Policy Threshold and Shares
+    if not n:
+        n = ALICE.n
+        if not force and not click.confirm(f'Use default value for N ({n})?', default=True):
+            n = click.prompt('Enter total number of shares (N)', type=click.INT)
+    if not m:
+        m = ALICE.m
+        if not force and not click.confirm(f'Use default value for M ({m})?', default=True):
+            n = click.prompt('Enter threshold (M)', type=click.IntRange(1, n+1))
+
+    # Policy Value
+    if not ALICE.federated_only and (not (bool(value) or bool(rate))):
         rate = ALICE.default_rate  # TODO #1709
-        click.confirm(f"Confirm default rate {rate}?", abort=True)
+        if not force:
+            use_default = click.confirm(f"Confirm default rate {rate}?", default=True)
+            if not use_default:
+                rate = click.prompt('Enter rate per period in Wei', type=WEI)
```

Comment:
Current rate range is in the GWEI order of magnitude, so better to use the `GWEI` type.

Reply:
:heavy_check_mark: